undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -129,12 +129,43 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         if (warMetaData == null) {
             return;
         }
-        String hostName = hostNameOfDeployment(warMetaData, defaultHost);
-        processDeployment(warMetaData, deploymentUnit, phaseContext.getServiceTarget(), hostName);
+        String deploymentName;
+        if (deploymentUnit.getParent() == null) {
+            deploymentName = deploymentUnit.getName();
+        } else {
+            deploymentName = deploymentUnit.getParent().getName() + "." + deploymentUnit.getName();
+        }
+        final UndertowService undertowService = (UndertowService) phaseContext.getServiceRegistry().getService(UndertowService.UNDERTOW).getValue();
+        final Set<Server> servers = undertowService.getServers();
```

Comment:
Dependency chain is added later to ensure deployment service is dependent on it, see https://github.com/ctomc/wildfly/blob/undertow-default-web/undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java#L316 
But at this point here we need to know if there is any default-web-module mapping or not. if it s than we process things bit different. if it isn't it goes to default server & host.

I really don't see any scenario where this would work wrongly, as servers & hosts are always started before any deployment can kick in. 
Even if host/server is added/changed at runtime it will be properly applied when next deployment is (re)deployed

Maybe only downside or non working scenario would be that when host would fail to start for whatever reason and deployment that should be bound to it's root context wouldn't be deployed there but to default host.
And as such would not fail deployment because of missing/non-started host. 
I think this is just fine as it is not a hard requirement asked by deployment (such is virtual-host/server setting in jboss-web.xml) but a soft hint driven from other side, so if host service is in error it cannot tell server what default web module it honors.